src/loader/README.md

```diff
@@ -16,6 +16,44 @@ version 0.8.0 will have timescaledb-0.8.0.so loaded.
    schedulers for any database that has TimescaleDB installed.
    This is done by a background task called the launcher.
 
+# Messages the launcher may receive
+The launcher implements a simple message queue to be notified when
+it should take certain actions, like starting or restarting a
+scheduler for a given database. Note that on startup of the server
+or on creation of a database, the launcher will attempt to start 
+a scheduler for each database, as it cannot figure out whether a 
+scheduler should exist for a given database. The scheduler is 
+responsible for shutting down if it should not exist.
+
+##Message types, when and why they are sent to the launcher:
+
+`start`: Used to start the scheduler by the user, it is meant to be
+an idempotent start, as in, if it is run multiple times, and startup
+is successful, it is the same as if it were run once. It is used
+mainly to reactivate a scheduler that the user had stopped.
+
+`stop`: Used to stop the scheduler in the case of a drop database 
+command, databases cannot be dropped if there are any open connections.
+The `stop` message is idempotent. 
+
+`restart`: Used to stop and restart the scheduler if it is running
```

Comment:
Lead with `may only be used to restart existing schedulers`? Right now it's buried in `if it is running`.

`The launcher will immediately restart the scheduler process, which will in turn wait for the txn that sent the message to finish. After the txn finishes, the scheduler will die if there is no longer an extension (as in the case of a successful `DROP EXTENSION`), or continue normally if there is a new version.`